client/src/pages/admin/CreateCategoryIntegration.test.js

```diff
@@ -0,0 +1,147 @@
+            await new categoryModel({ name: 'pens', slug: 'pens' }).save();
+            await new categoryModel({ name: 'phones', slug: 'phones' }).save();
+            await new categoryModel({ name: 'pants', slug: 'pants' }).save();
+
+            //ACT
+            render(
+                <AuthProvider>
+                <CartProvider>
+                    <SearchProvider>
+                        <Router>
+                            <Toaster />
+                            <CreateCategory />
+                        </Router>
+                    </SearchProvider>
+                </CartProvider>
+            </AuthProvider>
+            );
+
+            //ASSERT
+
+            //Admin Panel: mock replaced with actual component
+            await waitFor(() => {
+                expect(screen.getByText(/Admin Panel/)).toBeInTheDocument(); 
+            })
+            await waitFor(() => {
+                expect(screen.getByRole('link', { name: 'Create Category' })).toBeInTheDocument(); //link 1
+            })
+            await waitFor(() => {
+                expect(screen.getByRole('link', { name: 'Create Product' })).toBeInTheDocument(); //link 2
+            })   
+            await waitFor(() => {    
+                expect(screen.getByRole('link', { name: 'Products' })).toBeInTheDocument();
+            })
+            await waitFor(() => {    
+                expect(screen.getByRole('link', { name: 'Orders' })).toBeInTheDocument();
+            })
+
+            //Category Form: mock replaced with actual component
+
+            await waitFor(() => {
+                expect(screen.getByRole('textbox')).toBeInTheDocument();
+            })
+            
+            await waitFor(() => {
+                expect(screen.getByRole('textbox')).toHaveProperty('placeholder', "Enter new category");
+            })
+            await waitFor(() => {    
+                expect(screen.getByText('Submit')).toBeInTheDocument();
+            })
+
+            //Correctly displays fetched categories (existing)
+            await waitFor(() => {
+                expect(screen.getAllByText('dresses').length).toBeGreaterThan(0);
+            })
+            
+            await waitFor(() => {
+                expect(screen.getAllByText('books').length).toBeGreaterThan(0);
+            })
+            await waitFor(() => {    
+                expect(screen.getAllByText('toys').length).toBeGreaterThan(0);
+            })   
+            await waitFor(() => {
+                expect(screen.getAllByText('pens').length).toBeGreaterThan(0);
+            })
+            await waitFor(() => {    
+                expect(screen.getAllByText('pants').length).toBeGreaterThan(0);
+            })
+            await waitFor(() => {    
+                expect(screen.getAllByText('phones').length).toBeGreaterThan(0);
+            })
+
+            //categories not there should not be on screen 
```

Comment:
Hmmm, since 'socks' is not added in the database, wouldn't it confirm not appear? Thus, this check seems a bit redundant in my opinion haha

Reply:
Same for bottles

Reply:
True - Noted with thanks! 👍🏻